### PR TITLE
fix(spaces): user rxjs to fetch users

### PR DESCRIPTION
**Pull request by pmuir**

### CODE REVIEW

File: src/app/spaces/space.service.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { Injectable, Inject } from '@angular/core';
 import { Headers, Http, URLSearchParams } from '@angular/http';
 import { cloneDeep } from 'lodash';
 import { AuthenticationService, Logger, User, UserService } from 'ngx-login-client';
-import { Observable } from "rxjs";
+import { Observable } from 'rxjs';
```
Comment (joshuawilson) [inline src/app/spaces/space.service.ts]: While this works, it is better to be more specific.

`import { Observable } from 'rxjs/Observable';`

Comment (pmuir) [inline src/app/spaces/space.service.ts]: I've found that doesn't work very well as I loose certain operators etc.

Comment (joshuawilson) [inline src/app/spaces/space.service.ts]: hmm, ok. that is something I need to explore then

File: src/app/spaces/space.service.ts [hunk 2]
```diff
@@ -101,7 +102,20 @@ export class SpaceService {
           return newSpaces;
         }
       })
-      .catch((error) => { return this.handleError(error)});
+      .switchMap(val => {
+        let res: Observable<Space>[] = [];
```
Comment (joshuawilson) [inline src/app/spaces/space.service.ts]: `res` doesn't appear to be used. Is it needed?

Comment (pmuir) [inline src/app/spaces/space.service.ts]: good catch

File: src/app/spaces/space.service.ts [hunk 3]
```diff
@@ -110,58 +124,82 @@ export class SpaceService {
...
         // Extract data from JSON API response, and assert to an array of spaces.
-        let newSpaces: Space[] = response.json().data as Space[];
-        newSpaces.forEach((newSpace) => {
-          this.resolveUsersForSpace(newSpace);
-        });
-        return newSpaces;
+        return response.json().data as Space[];
+      })
+      .switchMap(val => {
+        let res: Observable<Space>[] = [];
```
Comment (joshuawilson) [inline src/app/spaces/space.service.ts]: another `res` not being referenced

**Pull request merged**